### PR TITLE
[Outreachy Task Submission]Phone Number and Email Validation Error Not Displayed in Notification Settings #4884

### PR DESCRIPTION
**Pull request by Appleeyes**
## Description

This pull request addresses the issue https://github.com/ushahidi/platform/issues/4884 where validation errors were not displayed correctly in the Notification Settings modal component. Previously, when users inputted an invalid email format or phone number format into the respective input fields, no validation error message was shown, leading to user confusion.

accountPhone

accountEmail

To address this issue, the validation logic for email and phone number inputs has been updated in the Notification Settings modal component. The email input field now correctly validates email formats using Angular's built-in Validators.email validator, while the phone number input field validates phone number formats using a custom regular expression pattern.

Additionally, the UI has been configured to display appropriate error messages next to the input fields. This enhancement improves the user experience by providing clear feedback to users about incorrect inputs, prompting them to correct the format accordingly.

phoneError

emailError

## Changes Made

- Updated the validation logic for email and phone number inputs in the Notification Settings modal component.
- Utilized Angular's Validators.email validator for email validation.
- Implemented a custom regular expression pattern for phone number validation.
- Configured the UI to display error messages next to the input fields when validation fails.

## Testing

- Manually tested the Notification Settings modal component.
- Inputted various email and phone number formats, including valid and invalid ones.
- Verified that error messages are displayed correctly for invalid inputs.
- Ensured that error messages disappear when valid inputs are provided.
- Tested the functionality across different browsers and screen sizes to ensure consistency.

This pull request resolves the issue of validation errors not being displayed correctly in the Notification Settings, enhancing the user experience and preventing potential issues arising from incorrect inputs.

### CODE REVIEW

File: apps/web-mzima-client/src/env.json
Comment (tuxpiper) [inline apps/web-mzima-client/src/env.json:1]: This change is not useful to others. @Appleeyes could you please remove it from the PR?

**Review by tuxpiper — CHANGES_REQUESTED** [review]
Very nice that you found the issue, documented it and solved it. 🙌 

Please take a look at my request for a small change. I think that's all it needs to be approved.

**Comment by tuxpiper** [comment]
@Angamanga this one is close to approval, just a tiny detail pending

File: apps/web-mzima-client/src/env.json
Comment (Appleeyes) [reply apps/web-mzima-client/src/env.json:1]: Thank you @tuxpiper I removed it.

**Review by tuxpiper — APPROVED** [review]
Approved :+1: Nice job, thank you!

**Pull request closed without merging**